### ISSUE TITLE
KeyError Issue Reason

### ISSUE THREAD

**Issue by nitinhariharan** [huggingface/transformers#30569]
### System Info

Hi Team
I am trying to predict for a fine tuned roberta base model for text classification task

from torch.utils.data import Dataset
from tqdm.auto import tqdm
from transformers import AutoTokenizer, pipeline
import torch.multiprocessing as mp
from fastfit import FastFit

tokenizer = AutoTokenizer.from_pretrained("location/checkpoint-9375/", truncation=True, max_length=512)
model = FastFit.from_pretrained("location/checkpoint-9375/").to('cuda')
pipe = pipeline("text-classification", model=model, tokenizer=tokenizer, device=0, truncation=True, max_length=512)

for out in pipe(dataset, batch_size=18):
print(out)

this is the prediction code and if i give batches it throws out key error but with one batch size it is running fine

![image](https://github.com/huggingface/transformers/assets/85015974/578ab626-2bf5-4f9f-9804-9886e1520237)
can you help me with it why it is so

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

from torch.utils.data import Dataset
from tqdm.auto import tqdm
from transformers import AutoTokenizer, pipeline
import torch.multiprocessing as mp
from fastfit import FastFit

tokenizer = AutoTokenizer.from_pretrained("location/checkpoint-9375/", truncation=True, max_length=512)
model = FastFit.from_pretrained("location/checkpoint-9375/").to('cuda')
pipe = pipeline("text-classification", model=model, tokenizer=tokenizer, device=0, truncation=True, max_length=512)

for out in pipe(dataset, batch_size=18):
print(out)

### Expected behavior

It should give the correspodent highest score label that is fine tuned with

**Comment by amyeroberts** [comment]
Hi @nitinhariharan, without access to the model or dataset, it's very hard for us to know what the issue is, as we can't replicate on our side. 

On the shared screenshot, there's an error message saying this model type is not supported by the text-classification pipeline
